File: rust/axum/src/modules/address/entity.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;
use chrono::NaiveDateTime;

#[derive(Serialize, Deserialize, Debug, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct AddressEntity {
    pub id: Option<Uuid>,
    pub logradouro: Option<String>,
    pub numero: Option<String>,
    pub bairro: Option<String>,
    pub cidade: Option<String>,
    pub uf: Option<String>,
    pub complemento: Option<String>,
    pub cep: Option<String>,
    #[sqlx(rename = "idCliente")]
    #[serde(rename = "idCliente")]
    pub id_cliente: Option<Uuid>,
    #[sqlx(rename = "dataDeCriacao")]
    #[serde(rename = "dataDeCriacao")]
    pub data_de_criacao: Option<NaiveDateTime>,
    #[sqlx(rename = "dataDeAtualizacao")]
    #[serde(rename = "dataDeAtualizacao")]
    pub data_de_atualizacao: Option<NaiveDateTime>,
}

File: go/fiber/internal/modules/address/entity/entity.go
package entity

import (
	"tcc/internal/types"

	"github.com/google/uuid"
)

type AddressEntity struct {
	ID              *uuid.UUID     `json:"id,omitempty"`
	Logradouro      string     		`json:"logradouro"`
	Numero          string     		`json:"numero"`
	Bairro          string     		`json:"bairro"`
	Cidade          string     		`json:"cidade"`
	UF              string     		`json:"uf"`
	Complemento     *string    		`json:"complemento,omitempty"`
	CEP             string     		`json:"cep"`
	IDCliente       *uuid.UUID     `json:"idCliente,omitempty"`
	DataDeCriacao   types.CustomTime  		`json:"dataDeCriacao"`
	DataDeAtualizacao types.CustomTime 	`json:"dataDeAtualizacao"`
}